crates/walrus-service/src/node.rs

```diff
@@ -1584,6 +1586,18 @@ impl StorageNode {
         // shards are created.
         let shard_map_lock = self.inner.storage.lock_shards().await;
 
+        // TODO(WAL-1040): Move this to a background task.
+        if self
+            .inner
+            .garbage_collection_config
+            .enable_blob_info_cleanup
+        {
+            self.inner
+                .storage
+                .process_expired_blob_objects(event.epoch)
```

Comment:
question: right now, this runs if the node is in recovery, which is fine during development. In the final version, I think maybe we can only run this if the node is in Active status, WDYT?